src/sgame/sg_bot_ai.cpp

```diff
@@ -1047,6 +1047,10 @@ AINodeStatus_t BotActionHeal( gentity_t *self, AIGenericNode_t *node )
 
 AINodeStatus_t BotActionSuicide( gentity_t *self, AIGenericNode_t* )
 {
+	//hopefully this is enough digits for all maps.
+	char str[] = "I'm stuck at x=0123456789 y=0123456789 z=0123456789 !?!";
+	snprintf( str, sizeof( str ), "I'm stuck at x=%-.0f y=%-.0f z=%-.0f !?!", self->s.origin[0], self->s.origin[1], self->s.origin[2] );
+	G_Say( self, SAY_TEAM, str );
```

Comment:
I'd use `G_LogPrintf` or `Log::Debug` here.

Generally speaking, `snprintf` seems fairly unidiomatic in Unvanquished and daemon. The usual way to do that is to use `va` or the newer `Str::Format`. Of course, with `G_LogPrintf` or `Log::Debug` you don't have this problem in the first place.

---

By the way, I think that you would benefit from defining a logger in the bot code. You can define this:

```C++
Log::Logger botLog = Log::Logger("bot", "", Log::Level::NOTICE);
```
Then use it like you would usually, except with `botLog.` instead of `Log::`. For example  `botLog.Debug("%i enemies", nbEnemies)`, `botLog.Warn("Unexepected something")`, `botLog.Notice("Some message")`

So that the day you want to debug stuff, you can change it to this:
```C++
Log::Logger botLog = Log::Logger("bot", "", Log::Level::DEBUG).WithoutSuppression();
```

(the `.WithoutSuppression()` is to disable the log spam rate-limiting.)

Reply:
coords are hardly readable, yes, and I would have preferred indication of the room, but I don't know how to get it. This, or to place a tag.

@necessarily-equal: as I said, I want players to move buildings which can cause such suicides, thus i think that, unlike for most messages which disappear way too fast anyway, this information can be useful.
Could be eggs that bots don't know how to move from, could be turrets or barricades placed so that they prevent bots to jump above them or steer around, etc. And it might not happen when someone is paying attention, too.